doc/source/composites.rst

```diff
@@ -2,48 +2,405 @@
+    >>> composite = compositor([filler, data_with_holes_1, data_with_holes_2,
+    ...                         data_with_holes_3])
+
+PaletteCompositor
+------------------
+
+:class:`PaletteCompositor` creates a color version of a single channel
+categorical dataset using a colormap::
+
+    >>> from satpy.composites import PaletteCompositor
+    >>> compositor = PaletteCompositor("palcomp")
+    >>> composite = compositor([local_scene['cma'], local_scene['cma_pal']])
+
+The palette should have a single entry for all the (possible) values
+in the dataset mapping the value to an RGB triplet.  Typically the
+palette comes with the categorical (e.g. cloud mask) product that is
+being visualized.
+
+DayNightCompositor
+------------------
+
+:class:`DayNightCompositor` merges two different composites.  The
+first composite will be placed on the day-side of the scene, and the
+second one on the night side.  The transition from day to night is
+done by calculating solar zenith angle (SZA) weighed average of the
+two composites.  The SZA can optionally be given as third dataset, and
+if not given, the angles will be calculated.  Width of the blending
+zone can be defined when initializing the compositor (default values
+shown in the example below).
+
+    >>> from satpy.composites import DayNightCompositor
+    >>> compositor = DayNightCompositor("dnc", lim_low=85., lim_high=95.)
+    >>> composite = compositor([local_scene['true_color'],
+    ...                         local_scene['night_fog']])
+
+RealisticColors
+---------------
+
+:class:`RealisticColors` compositor is a special compositor that is
+used to create realistic near-true-color composite from MSG/SEVIRI
+data::
+
+    >>> from satpy.composites import RealisticColors
+    >>> compositor = RealisticColors("realcols", lim_low=85., lim_high=95.)
+    >>> composite = compositor([local_scene['VIS_006'],
+    ...                         local_scene['VIS_008'],
+    ...                         local_scene['HRV']])
+
+CloudCompositor
+---------------
+
+:class:`CloudCompositor` can be used to threshold the data so that
+"only" clouds are visible.  These composites can be used as an overlay
+on top of e.g. static terrain images to show a rough idea where there
+are clouds.  The data are thresholded using three variables::
 
-One important thing to notice is that there is an internal difference between a composite and an image. A composite
-is defined as a special dataset which may have several bands (like R, G, B bands). However, the data isn't stretched,
-or clipped or gamma filtered until an image is generated.
+ - ``transition_min``: values below or equal to this are clouds -> opaque white
+ - ``transition_max``: values above this are cloud free -> transparent
+ - ``transition_gamma``: gamma correction applied to clarify the clouds
 
+Usage (with default values)::
+
+    >>> from satpy.composites import CloudCompositor
+    >>> compositor = CloudCompositor("clouds", transition_min=258.15,
```

Comment:
This has the same problem as `DayNightCompositor`; there is no positional argument (see [my comment there](https://github.com/pytroll/satpy/pull/513/files#r273089889)):

https://github.com/pytroll/satpy/blob/dff69ad5dc8373961111b4e1956f83e92a68618c/satpy/composites/__init__.py#L1135

so this fails again with `TypeError: __init__() got multiple values for argument 'transition_min'`.